src/main/kotlin/com/vk/kphpstorm/inspections/KphpNotSupportFunctionsInspection.kt

```diff
@@ -0,0 +1,48 @@
+package com.vk.kphpstorm.inspections
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElementVisitor
+import com.jetbrains.php.lang.inspections.PhpInspection
+import com.jetbrains.php.lang.psi.elements.FunctionReference
+import com.jetbrains.php.lang.psi.elements.PhpReference
+import com.jetbrains.php.lang.psi.visitors.PhpElementVisitor
+import com.vk.kphpstorm.inspections.quickfixes.ReplaceToKphpFunctionsQuickFix
+
+class KphpNotSupportFunctionsInspection : PhpInspection() {
+    companion object {
+        private val PHP_TO_KPHP_FUNCTIONS = mapOf(
+            "end" to listOf("array_last_value", "array_last_key"),
+            "reset" to listOf("array_first_value", "array_first_key"),
+        )
+    }
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : PhpElementVisitor() {
+            override fun visitPhpFunctionCall(reference: FunctionReference) {
+                if (reference.immediateNamespaceName != "")
```

Comment:
Why not just:

```kotlin
val functionName = reference.fqn
```

And:

```kotlin
private val PHP_TO_KPHP_FUNCTIONS = mapOf(
  "\\end" to listOf("array_last_value", "array_last_key"),
  "\\reset" to listOf("array_first_value", "array_first_key"),
)
```

This seems like a simpler solution and doesn't require any reference resolutions.

Reply:
Thanks to this code, we do not receive incorrect warnings:
```kotlin
if (reference.immediateNamespaceName != "" && reference.immediateNamespaceName != "\\")
    return

if ((reference as PhpReference).resolveLocal().isNotEmpty())
    return
```

```php
<?php # functions.php

namespace MyFunctions;

function reset(): string {
    return "sddsd";
}

$animal = array('dog', 'cat', 'fox');

echo reset(); // not warning 'KPHP does not support'
echo end($animal); // warning 'KPHP does not support'
echo \N\end($animal); // not warning 'KPHP does not support'
echo \reset($animal); // warning 'KPHP does not support'
```

```php
<?php # index.php

namespace N;

use MyFunctions;

function end(object|array &$array): string {
    return "";
}

$fruits = array('apple', 'banana', 'cranberry');

echo end($fruits); // not warning 'KPHP does not support'
echo MyFunctions\reset(); // not warning 'KPHP does not support'
echo reset($fruits); // warning 'KPHP does not support'
echo \reset($fruits); // warning 'KPHP does not support'
echo \end($fruits); // warning 'KPHP does not support'
```

```php
<?php # example.php

use function  N\end;
use function  N\end as reset;

$a = [];

echo end($a); // not warning 'KPHP does not support'
echo reset($a); // not warning 'KPHP does not support'
echo \reset($a); // warning 'KPHP does not support'
echo \end($a); // warning 'KPHP does not support'
```

Reply:
Yep, it's true